atomic_reactor/util.py

```diff
@@ -738,6 +738,41 @@ def __init__(self, registry, insecure=False, dockercfg_path=None, access=None):
 
         self.session = get_retrying_requests_session()
 
+    @classmethod
+    def create_from_config(cls, registry, workflow, access=None):
+        """
+        Create a session for the specified registry based on configuration in reactor config map.
+
+        If the registry is configured in pull_registries, use that configuration. Otherwise,
+        use the default source_registry configuration (if present, else an error is thrown).
+
+        :param registry: str, registry to create session for (as hostname[:port], no http(s))
+        :param workflow: DockerBuildWorkflow, contains configuration for registries
+        :param access: List of actions this session is allowed to perform, e.g. ('push', 'pull')
+
+        :return: RegistrySession for the specified registry
+        """
+        # Has to be imported here (cyclic import)
+        from atomic_reactor.plugins.pre_reactor_config import (get_source_registry,
```

Comment:
Keep it for now here, we can move it to custom submodule (or resolve cyclic import in different way) in another PR if needed.

Reply:
Or decouple workflow from it, only ask for default and pull_registries configuration in params

Reply:
I tried the decoupling solution, it might be kind of hard to use, you would have to do this before creating a registry session:
```python
source_registry = get_source_registry(workflow, fallback=None)
pull_registries = get_pull_registries(workflow, fallback=[])
```
Maybe the cyclic import is fine for now, IDK

Reply:
I'm thinking how much ugly it would be to add sort of registry factory method into workflow. It will encapsulate everything, hides workflow/configuration implementation, and can cache instances :thinking:

Reply:
keep it with cyclic imports for now
functionality first, cleanup later :D